interactions/client/auto_shard_client.py

```diff
@@ -35,6 +35,8 @@ def __init__(self, *args, **kwargs) -> None:
         self.auto_sharding = "total_shards" not in kwargs
         super().__init__(*args, **kwargs)
 
+        self.only_shards: Optional[List[int]] = kwargs.get("only_shards", None)
```

Comment:
Why not use `Optional[list[int]]` and avoid an import?

Reply:
Not sure how I feel about the naming either - it's rather confusing. Why not just name it `shard_ids`?

Reply:
> Not sure how I feel about the naming either - it's rather confusing. Why not just name it `shard_ids`?

I didn't want anyone to confuse it with "shard_id." But either way is fine and I can change it if you feel only_shards is confusing.

Reply:
> Why not use `Optional[list[int]]` and avoid an import?

This is nitpicking, we use "List" hundreds of times in other files, I believe that if I am using the Typing library for Optional, I might as well use it for List as well.

Reply:
> I didn't want anyone to confuse it with "shard_id." But either way is fine and I can change it if you feel only_shards is confusing.

`only_shards` feels even more confusing than `shard_ids` honestly. Perhaps we should look into renaming attributes in `AutoShardedClient` in general, though that's for a different PR.